src/js/core/directives/ui-grid.js

```diff
@@ -24,20 +24,23 @@
       $scope.grid = self.grid;
 
       if ($attrs.uiGridColumns) {
-        $attrs.$observe('uiGridColumns', function(value) {
-          self.grid.options.columnDefs = value;
+        deregFunctions.push( $attrs.$observe('uiGridColumns', function(value) {
+          self.grid.options.columnDefs = angular.isString(value) ? angular.fromJson(value) : value;
```

Comment:
While writing unit tests, I realized there was a mistake with this watch. Since this is an attribute watch, the value of it might be a string that needs to be converted to an object.